### ISSUE TITLE
Relevant new PyMC3 features

### ISSUE THREAD

**Issue by twiecki** [bambinos/bambi#55]
Not sure how closely you guys follow PyMC3 development but maybe two things would be of interest:

* Smart initialization: `pm.sample(2000)` on continuous models will use ADVI to estimate a good mass matrix for NUTS to speed things up and help with bad intialization. Should be the preferred method.
* A new `UserModel` that can be inherited, see: https://github.com/pymc-devs/pymc3/pull/1525, specifically https://github.com/pymc-devs/pymc3/pull/1525/files#diff-f85ae0776100e0962d318aaad0c5340fR90

**Comment by tyarkoni** [comment]
Thanks for flagging these changes, I hadn't seen them. (I try to keep track of what's going on in PyMC3, but you guys move way too fast!)

Re: smart initialization, one thing we've noticed in the past is that with large models that include many random effects, initializing based on ADVI or another short NUTS run sometimes results in the sampler stalling very quickly, whereas it doesn't stall without smart initialization. I see pymc-devs/pymc3#555 was closed without any clear resolution; is this something that we have any reason to think has been addressed by any changes? I guess we can try out the new approach and see what happens.

The `UserModel` looks useful. I think we're already doing a lot of that in our `Model` class, which is deliberately abstracted away from PyMC3 (so we can potentially add other backends later, e.g., Stan), but we might be able to leverage this to dynamically initialize new distributions more cleanly within the `PyMC3BackEnd`. We'll take a look.
